### PR TITLE
Initilize variables in scene\2d

### PR DESCRIPTION
**Pull request by Kyetuur**
Initialization of variables in scene\2d
#52674

### CODE REVIEW

**Comment by KoBeWi** [comment]
Built-in Godot types like Vector2 or Color don't need to be initialized. Only primitive types like ints, bools, pointers etc.

**Comment by Kyetuur** [comment]
Ok, i will change it

**Comment by Kyetuur** [comment]
@KoBeWi I reverted changes on unprimitive members

**Comment by KoBeWi** [comment]
You are using `f` suffix inconsistently, I think you should just remove it.
Other than that the changes are fine (I didn't check if it covers all fields though).

File: scene/2d/area_2d.h [hunk 1]
```diff
@@ -49,7 +49,7 @@ class Area2D : public CollisionObject2D {
 private:
 	SpaceOverride space_override = SPACE_OVERRIDE_DISABLED;
 	Vector2 gravity_vec;
-	real_t gravity = 0.0;
+	real_t gravity = 0;
```
Comment (KoBeWi) [inline scene/2d/area_2d.h]: This change is unnecessary (especially considering that below there are still some `0.0`s).

File: scene/2d/gpu_particles_2d.h [hunk 2]
```diff
@@ -47,17 +47,17 @@ class GPUParticles2D : public Node2D {
...
+	double pre_process_time = 0;
+	real_t explosiveness_ratio = 0;
+	real_t randomness_ratio = 0;
+	double speed_scale = 0;
 	Rect2 visibility_rect;
-	bool local_coords;
-	int fixed_fps;
-	bool fractional_delta;
+	bool local_coords = false;
+	int fixed_fps = false;
```
Comment (KoBeWi) [inline scene/2d/gpu_particles_2d.h:59]:
```suggestion
	int fixed_fps = 0;
```

**Comment by akien-mga** [comment]
Sorry for the delay reviewing. This would need to be redone from scratch as there are conflicts on most changed files (the initializations may have been added by other PRs in the meantime).

**Pull request closed without merging**